### PR TITLE
feat: improve OTP Delete keydown

### PR DESCRIPTION
**Pull request by Wxh16144**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

按下键盘Delete 时应该删除后一项？

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      --     |
| 🇨🇳 Chinese |       --    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by Wxh16144** [comment]
https://preview-48866-ant-design.surge.sh/~demos/components-input-demo-otp

File: components/input/OTP/index.tsx [hunk 1]
```diff
@@ -249,6 +249,7 @@ const OTP = React.forwardRef<OTPRef, OTPProps>((props, ref) => {
               onActiveChange={onInputActiveChange}
               autoFocus={index === 0 && autoFocus}
               {...inputSharedProps}
+              __internalLastIndex={length - 1}
```
Comment (afc163) [inline components/input/OTP/index.tsx:252]: 这个 prop 命名很奇怪。

**Comment by Wxh16144** [comment]
有点曲解了设计，现在每个 input 其实是 autoselect 的。 后面看看有没有其他人有类似的需求吧～

**Comment by teamco** [comment]
It seems like there is also a validation problem:
1. e.g. If you fill in all the fields and then start erasing, then the validation does not work
2. error message not working as expected: {min: min,message: t(intl, 'form.validation.min', { entity: _label, min })}

**Pull request closed without merging**